std/experimental/allocator/building_blocks/bitmapped_block.d

```diff
@@ -1216,13 +1223,13 @@ struct BitmappedBlockWithInternalPointers(
     {
         auto r = _heap.allocate(bytes);
         if (!r.ptr) return r;
-        immutable block = (r.ptr - _heap._payload.ptr) / _heap.blockSize;
+        immutable block = (() @trusted => (r.ptr - _heap._payload.ptr) / _heap.blockSize)();
         immutable blocks =
             (r.length + _heap.blockSize - 1) / _heap.blockSize;
         if (!ensureRoomForAllocStart(block + blocks))
         {
             // Failed, free r and bailout
-            _heap.deallocate(r);
+            () @trusted { _heap.deallocate(r); }();
```

Comment:
Nullify `r` in the `@trusted` section.

Reply:
For what purpose?

Reply:
Otherwise `r` is a dangling pointer in `@safe` code. Can't have dangling pointers in `@safe` code.